### PR TITLE
SLF4JBridgeHandler.Initializer

### PR DESCRIPTION
**Pull request by mfulton26**

### CODE REVIEW

**Comment by mfulton26** [comment]
Refreshed. Thanks.

File: jul-to-slf4j/src/main/java/org/slf4j/bridge/SLF4JBridgeHandler.java [hunk 1]
```diff
@@ -300,4 +313,26 @@ public void publish(LogRecord record) {
...
+     */
+    public static class Initializer {
+
+        /**
+         * Registers {@link SLF4JBridgeHandler} as handler for the j.u.l. root logger.
+         *
+         * @throws IOException if there are problems reading from the stream.
+         */
+        public Initializer() throws IOException {
+            String loggingConfigurationString = "handlers = " + SLF4JBridgeHandler.class.getName();
```
Comment (mfulton26) [inline jul-to-slf4j/src/main/java/org/slf4j/bridge/SLF4JBridgeHandler.java]: I've added a test locally and now I realize that I don't need to create an in-memory configuration file to read. I will push an update in a bit.

**Comment by mfulton26** [comment]
Unit test added and main code simplified. Please let me know if any changes are needed. This should be ready for the next release. Thanks.

File: jul-to-slf4j/src/test/java/org/slf4j/bridge/SLF4JBridgeHandlerTest.java [hunk 2]
```diff
@@ -178,4 +179,10 @@ void assertLevel(int index, org.apache.log4j.Level expectedLevel) {
         LoggingEvent le = (LoggingEvent) listAppender.list.get(index);
         assertEquals(expectedLevel, le.getLevel());
     }
+
+    public void testInitializer() throws IOException {
```
Comment (mfulton26) [inline jul-to-slf4j/src/test/java/org/slf4j/bridge/SLF4JBridgeHandlerTest.java]: This no longer needs `throws`...

Comment (mfulton26) [inline jul-to-slf4j/src/test/java/org/slf4j/bridge/SLF4JBridgeHandlerTest.java]: Updated and ready for review.

**Comment by mfulton26** [comment]
I had to add back logic not using `LogManager.getLogManager().getLogger("")` because the root logger, "", is not yet initialized when a j.u.l. configuration class is constructed (it causes a `NullPointerException`).

Ready for review.

**Comment by anthavio** [comment]
Why oh why is this still not merged in 2020 so I have to copy this again...

**Comment by mfulton26** [comment]
@ceki, any concerns with merging this?

@anthavio I don't actively work in Java these days but if you or someone else would be willing to update this PR (I see the unit test file has some conflict(s) now) I am confident that @ceki would be willing to approve and merge.

**Comment by anthavio** [comment]
OK, I will try to refresh this PR to mergeable state

**Comment by ceki** [comment]
Please create a Jira issue for this PR.

**Comment by anthavio** [comment]
https://jira.qos.ch/browse/SLF4J-489

**Pull request closed without merging**